mypy/test/data/check-inference.test

```diff
@@ -1415,13 +1415,16 @@ main: note: In member "f" of class "A":
 [case testAttributePartiallyInitializedToNoneWithMissingAnnotation]
 class A:
     def f(self) -> None:
-        self.x = None # E: Need type annotation for variable
+        self.x = None
```

Comment:
With strict optional checking enabled, `None` will be inferred as a type if no other information is available other than the None assignment (and this is already tested).